lib/src/components/base/base.dart

```diff
@@ -1,7 +1,20 @@
 import '../../gen/common/v1/common.pb.dart';
+import '../../gen/component/base/v1/base.pb.dart';
 import '../../resource/base.dart';
 import '../../robot/client.dart';
 
+class BaseProperties {
+  double turningRadiusMeters;
+  double widthMeters;
+  double wheelCircumferenceMeters;
+
+  BaseProperties(this.turningRadiusMeters, this.widthMeters, this.wheelCircumferenceMeters);
+
+  factory BaseProperties.fromProto(GetPropertiesResponse pbResponse) {
+    return BaseProperties(pbResponse.turningRadiusMeters, pbResponse.widthMeters, pbResponse.wheelCircumferenceMeters);
+  }
+}
+
```

Comment:
Instead of creating a class that mirrors the existing type, you can use a `typedef` similar to [here](https://github.com/viamrobotics/viam-flutter-sdk/blob/82dbe48c1a26e28fe6ab6e195a1e6dea1bb7d9b1/lib/src/components/power_sensor/power_sensor.dart#L7)

Reply:
Yeah will do, but wondering why you want this Properties type to be `typedef`? There are only two other properties (motor and camera) but they are both set up as classes

Reply:
Great question -- likely because we did those earlier without realizing we didn't need to create custom classes for it. Those should actually be swapped to `typdef` as well.

IMO we should use `typedef` because they will automatically stay in sync with the protos should anything changed/added/removed

Reply:
do you want me to change the other two while I'm at it or would you rather they be done separately?

Reply:
I mean, if you want to do it that'll save some time and an additional PR! Thanks :)